File: python/meter-health/csv_formatter.py
# -*- coding: utf-8 -*-
import csv
import config.credentials

class CsvFormatter:
    def __init__(self,headers):
        self.file_name = config.credentials.file_name
        self.headers = headers
        self.create_file_with_headers()

    def create_file_with_headers(self):
        with open(self.file_name, 'w+') as csv_file:
            writer = csv.writer(csv_file)
            writer.writerow(self.headers)

    def write(self,info):
        with open(self.file_name, 'a+') as csv_file:
            writer = csv.writer(csv_file)
            writer.writerow(info)

File: ruby/meter-health/csv_formatter.rb
require 'csv'
require_relative 'helper_methods.rb'

class CsvFormatter

  include HelperMethods

  def initialize(headers)
    @headers = headers
    @credentials = retrieve_credentials
    create_file_with_headers
  end

  def write(info)
    CSV.open(@credentials["file_name"], "a+") do |csv|
      csv << info
    end
  end

  private

  def create_file_with_headers
    CSV.open(@credentials["file_name"], "w+") do |csv|
      csv << @headers
    end
  end
end